src/viking.js

```diff
@@ -1,11 +1,124 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength){
+        this.health = health;
+        this.strength = strength;
+    
+    }
+    attack(){
+        return this.strength        
+    }
+    receiveDamage(damage){
+        this.health -= damage;
+        
+    }
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength);
+        this.name = name    
+    }
+
+    receiveDamage(damage){
+
+    super.receiveDamage(damage);
+    
+    if(this.health > 0){     
+
+    return `${this.name} has received ${damage} points of damage`
+    
+    }else if (this.health <= 0 ){
+
+        return `${this.name} has died in act of combat`
+        
+    }
+}
+battleCry(){
+
+    return "Odin Owns You All!"
+
+}
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    receiveDamage(damage){
+
+        super.receiveDamage(damage)
+
+        if(this.health > 0){
+            return `A Saxon has received ${damage} points of damage` ;
+        } else if(this.health <= 0){
+            return `A Saxon has died in combat`;
+        }
+    }
+}
+
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+    saxonArmy = [];
+
+    addViking(Viking){
+        this.vikingArmy.push(Viking);
+    }
+    addSaxon(Saxon){
+        this.saxonArmy.push(Saxon);
+    }
+    vikingAttack(){
+        let randomViking = this.vikingArmy[Math.floor(Math.random() * (this.vikingArmy.length))];
+        let randomSaxonInd = Math.floor(Math.random () * (this.saxonArmy.length));
+        let randomSaxon = this.saxonArmy[randomSaxonInd];
+        let receiveDamageOfSaxon = this.randomSaxon.receiveDamage(this.randomViking.strength)
+
+        if(randomSaxon.health <= 0){
+           this.saxonArmy.splice(randomSaxonInd,1);
+        }
+        
+        return receiveDamageOfSaxon
+    }
+
+    saxonAttack(){
+        let randomSaxon = this.saxonArmy[Math.floor(Math.random() * (this.saxonArmy.length))] ;
+        let randomVikingInd = Math.floor(Math.random() * (this.vikingArmy.length));
+        let randomViking = this.vikingArmy[randomVikingInd];
+        let receiveDamageOfViking = this.randomViking.receiveDamage(this.randomSaxon.strength);
```

Comment:
Here you need to remove the `this` from `randomViking` and `randomSaxon` because you're creating those inside the method, so you don't need to access them with the `this` keyword, in that case the method is trying to access those variable from the object (war) scope and not the ones that you're creating inside the method